tests/phase0/test_sanity.py

```diff
@@ -140,6 +141,32 @@ def test_proposer_slashing(state):
     return state, [block], test_state
 
 
+def test_attester_slashing(state):
+    test_state = deepcopy(state)
+    attester_slashing = get_valid_attester_slashing(state)
+    validator_index = attester_slashing.attestation_1.custody_bit_0_indices[0]
+
+    #
+    # Add to state via block transition
+    #
+    block = build_empty_block_for_next_slot(test_state)
+    block.body.attester_slashings.append(attester_slashing)
+    state_transition(test_state, block)
+
+    assert not state.validator_registry[validator_index].initiated_exit
+    assert not state.validator_registry[validator_index].slashed
+
+    slashed_validator = test_state.validator_registry[validator_index]
+    assert not slashed_validator.initiated_exit
+    assert slashed_validator.slashed
+    assert slashed_validator.exit_epoch < spec.FAR_FUTURE_EPOCH
+    assert slashed_validator.withdrawable_epoch < spec.FAR_FUTURE_EPOCH
+    # lost whistleblower reward
+    assert get_balance(test_state, validator_index) < get_balance(state, validator_index)
```

Comment:
Maybe also check `proposer_index`? It happened to me when pseudo-randomly, `proposer == slashed_validator` in unit tests. 😆